resources/default.rb

```diff
@@ -13,10 +13,18 @@
 property :service_name, kind_of: String, default: lazy { instance_name }
 property :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
 property :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+property :api_endpoint, kind_of: String, default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_wait, kind_of: Integer, default: lazy { node['nexus3']['api']['wait'] }
 
 action :install do
   install_dir = ::File.join(new_resource.path, "nexus-#{version}")
 
+  node.run_state[:initial_install] = false
+  # This flag determines if we can change the admin password.
+  unless ::File.exist?(::File.join(new_resource.data, 'tmp'))
```

Comment:
Each time I want to change the password the resource won't be trigger will it? The `tmp` directory will be present.

Reply:
This is only about changing the password for the admin user, which is used for all API calls. If you want to change it at another time, you will need to pass the previous password to the resource, and it will not be trivial (or update the in-memory data structure in Chef, which is generally frowned upon).

Other attempts at this: